### PR TITLE
Validate object/interfaces implement all transitive interfaces

### PR DESCRIPTION
**Pull request by maartenvanvliet**
See https://github.com/absinthe-graphql/absinthe/issues/1080 for a full description

 * adds a type reference check to see whether interfaces on interfaces exist
 * adds a check to ensure that an object/interface implements all interfaces, also interfaces it transitively has

@tlvenn can you take a look

fixes https://github.com/absinthe-graphql/absinthe/issues/1080

### CODE REVIEW

File: test/absinthe/schema/rule/object_must_implement_interfaces_test.exs [hunk 1]
```diff
@@ -22,6 +23,10 @@ defmodule Absinthe.Schema.Rule.ObjectMustImplementInterfacesTest do
     interface :parented do
       field :parent, :named
       field :another_parent, :named
+
+      resolve_type fn
+        _, _ -> nil
```
Comment (tlvenn) [inline test/absinthe/schema/rule/object_must_implement_interfaces_test.exs:28]: Is there any reason why the resolve_type does not include `dog`, `cat` or `user` as in the other interfaces ?
Like:
```
resolve_type fn
  %{type: :dog}, _ -> :dog
  %{type: :user}, _ -> :user
  %{type: :cat}, _ -> :cat
  _, _ -> nil
end
```

**Review by tlvenn — COMMENTED** [review]
Looks good to me, thanks for this @maartenvanvliet !

File: test/absinthe/schema/rule/object_must_implement_interfaces_test.exs (hunk 1 above)
Comment (maartenvanvliet) [reply test/absinthe/schema/rule/object_must_implement_interfaces_test.exs:28]: Forgot about that one 🙂

Comment (tlvenn) [reply test/absinthe/schema/rule/object_must_implement_interfaces_test.exs:28]: Hehe no worry, as soon as you get the chance to fix this I will merge this PR, thanks again.

Comment (maartenvanvliet) [reply test/absinthe/schema/rule/object_must_implement_interfaces_test.exs:28]: 👍

**Pull request merged**